graphsync.go

```diff
@@ -5,14 +5,23 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/google/uuid"
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-ipld-prime"
 	"github.com/ipld/go-ipld-prime/traversal"
 	"github.com/libp2p/go-libp2p-core/peer"
 )
 
 // RequestID is a unique identifier for a GraphSync request.
-type RequestID int32
+type RequestID uuid.UUID
+
+func (rid RequestID) String() string {
+	return uuid.UUID(rid).String()
```

Comment:
Question for Go ppl: why can't I just call `String()` on a `RequestID` object if it's of type `uuid.UUID`? Why is this indirection necessary where `uuid.UUID` is in the `type`?